examples/taskruns/workspace-readonly.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: my-pvc-3
+spec:
+  resources:
+    requests:
+      storage: 5Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TaskRun
+metadata:
+  generateName: workspaces-readonly-
+spec:
+  workspaces:
+    - name: write-allowed
+      persistentVolumeClaim:
+        claimName: my-pvc-3
+    - name: write-disallowed
+      persistentVolumeClaim:
+        claimName: my-pvc-3
+  taskSpec:
+    workspaces:
+    - name: write-allowed
+    - name: write-disallowed
+      readOnly: true
+    steps:
+    - name: write-allowed
+      image: ubuntu
+      script: echo "hello" > $(workspaces.write-allowed.path)/foo
+    - name: read-allowed
+      image: ubuntu
+      script: cat $(workspaces.write-allowed.path)/foo | grep "hello"
+    - name: write-disallowed
+      image: ubuntu
+      script: echo "goodbye" > $(workspaces.write-disallowed.path)/foo || true
+    - name: read-disallowed
+      image: ubuntu
+      script: cat $(workspaces.write-disallowed.path)/foo | grep "hello"
```

Comment:
Is read supposed to be disallowed? If so would `grep "hello"` fail?

Reply:
oops, that's confusing.  I simply meant that the read-disallowed step runs after the write-disallowed.  Reading from the volume isn't supposed to be disallowed.  I'll update the name here.